.github/workflows/lint.yaml

```diff
@@ -85,13 +85,18 @@ jobs:
         key: macos-12-build-cache-${{ hashFiles('./scripts/builder_setup.sh') }}
 
     - name: Clone datadog-agent repository
+      uses: nick-fields/retry@v3
       env:
         VERSION: ${{ github.event.inputs.datadog_agent_ref || env.DEFAULT_DATADOG_AGENT_REF }}
-      run: |
-        bash ./scripts/clone_agent.sh
-        cd $HOME/go/src/github.com/DataDog/datadog-agent
-        export VERSION_CACHE_CONTENT="${{github.event.inputs.version_cache}}"
-        if [ ! -z "$VERSION_CACHE_CONTENT" ]; then base64 -d <(echo "$VERSION_CACHE_CONTENT") > ./agent-version.cache; fi
+      with:
+        max_attempts: 3
+        timeout_minutes: 10
```

Comment:
Is there a specific reason to put 10 minutes here? That seems a lot given that the clone should be pretty fast